providers/threeplay.go

```diff
@@ -73,15 +73,43 @@ func (c *ThreePlayProvider) GetProviderJob(id string) (*database.ProviderJob, er
 	return providerJob, nil
 }
 
-// DispatchJob sends a video file to 3play for transcription and captions generation
+// DispatchJob sends a video file to 3play for transcription and captions generation or generates a expiring editing link
+// when the media_file_url param is provided
 func (c *ThreePlayProvider) DispatchJob(job *database.Job) error {
 	jobLogger := c.logger.WithFields(log.Fields{"JobID": job.ID, "Provider": job.Provider})
+	// Review job route
+	if mediaFileID, ok := job.ProviderParams["media_file_id"]; ok {
+		hoursInt := 2
+		var err error
+		if hoursUntilExpiration, ok := job.ProviderParams["hours_until_expiration"]; ok {
+			hoursInt, err = strconv.Atoi(hoursUntilExpiration)
+			if err != nil {
+				jobLogger.WithError(err).Error("Could not convert hours until expiration")
+			}
+		}
+
+		reviewURL, err := c.GetEditingLink(mediaFileID, hoursInt)
+		if err != nil {
+			jobLogger.WithError(err).Error("Could not generate review url")
+			return err
+		}
+
+		job.ProviderParams["ProviderID"] = mediaFileID
+		job.ProviderParams["ReviewURL"] = reviewURL
+		return nil
+	}
+
+	// Creation job route
 	query := url.Values{}
 	turnaroundLevel := "asr"
+	callbackURL := ""
 	for k, v := range job.ProviderParams {
-		if k == "turnaround_level_id" {
+		switch k {
+		case "turnaround_level_id":
 			turnaroundLevel = v
-		} else {
+		case "callback_url":
```

Comment:
If I understand this correctly, a service downstream of video-captions-api is setting the ``callback_url``, which is a URL to the captions-api itself that then gets forwarded to that downstream service (configured with the environment variable ``CALLBACK_URL``).

Could the ``callback_url`` be set to a URL that can be interpreted directly by the downstream service? This way, ProcessCallback and ``CALLBACK_URL`` wouldn't be needed.

If that's not possible (because the downstream service doesn't have a reference yet to the job it is creating?), could video-captions-api include the jobId into the provided callback_url as a query string parameter? Then you don't need ``GetJobByProviderID`` (nor the datastore index).

Reply:
I didn't want the downstream service to receive the callback directly from 3play because they have no knowledge of it as a captions provider. My hunch was to keep 3play-related things inside the captions api.
I do like the idea of adding the jobID as the query string parameter, though! I'll add that. It would simplify things, though we might still want to check for it when processing the callback in case it goes missing somehow 😱